napari/plugins/__init__.py

```diff
@@ -1,409 +1,11 @@
-        warn(
-            f'Plugin {plugin_name!r} provided a non-dict object to '
-            f'{hook_name!r}: data ignored.'
-        )
-        return
-
-    _data = {}
-    for name, datum in list(data.items()):
-        if isinstance(datum, dict):
-            if 'data' not in datum or 'display_name' not in datum:
-                warn(
-                    f'In {hook_name!r}, plugin {plugin_name!r} provided an '
-                    f'invalid dict object for key {name!r} that does not have '
-                    'required keys: "data" and "display_name".  Ignoring'
-                )
-                continue
-        else:
-            datum = {'data': datum, 'display_name': name}
-
-        if not (
-            callable(datum['data']) or isinstance(datum['data'], (str, Path))
-        ):
-            warn(
-                f'Plugin {plugin_name!r} provided invalid data for key '
-                f'{name!r} in the dict returned by {hook_name!r}. '
-                f'(Must be str, callable, or dict), got ({type(datum["data"])}).'
-            )
-            continue
-        _data[name] = datum
-
-    if plugin_name not in _sample_data:
-        _sample_data[plugin_name] = {}
-
-    _sample_data[plugin_name].update(_data)
-
-
-def discover_dock_widgets():
-    """Trigger discovery of dock_widgets plugins"""
-    dw_hook = plugin_manager.hook.napari_experimental_provide_dock_widget
-    dw_hook.call_historic(result_callback=register_dock_widget, with_impl=True)
-    fw_hook = plugin_manager.hook.napari_experimental_provide_function
-    fw_hook.call_historic(
-        result_callback=register_function_widget, with_impl=True
-    )
-
-
-def discover_sample_data():
-    """Trigger discovery of sample data."""
-    sd_hook = plugin_manager.hook.napari_provide_sample_data
-    sd_hook.call_historic(result_callback=register_sample_data, with_impl=True)
-
-
-def available_samples() -> Tuple[Tuple[str, str], ...]:
-    """Return a tuple of sample data keys provided by plugins.
-
-    Returns
-    -------
-    sample_keys : Tuple[Tuple[str, str], ...]
-        A sequence of 2-tuples ``(plugin_name, sample_name)`` showing available
-        sample data provided by plugins.  To load sample data into the viewer,
-        use :meth:`napari.Viewer.open_sample`.
-
-    Examples
-    --------
-
-    .. code-block:: python
-
-        from napari.plugins import available_samples
-
-        sample_keys = available_samples()
-        if sample_keys:
-            # load first available sample
-            viewer.open_sample(*sample_keys[0])
-    """
-    return tuple((p, s) for p in _sample_data for s in _sample_data[p])
-
-
-discover_sample_data()
-
-
-def load_settings_plugin_defaults(SETTINGS):
-    """Sets SETTINGS plugin defaults on start up from the defaults saved in
-    the plugin manager.
-    """
-
-    SETTINGS._defaults['plugins'].call_order = plugin_manager.call_order()
-
+plugin_manager = NapariPluginManager()
+plugin_manager._initialize()
```

Comment:
Love that all this stuff disappears in this file!! One question - why is there a separate `_initialize` method here, instead of everything just being in the init?

Reply:
this was an idea I was playing with related to [testing]... so that you can instantiate a plugin manager without our builtins... but we could also make a special subclass just for testing that defeats  `_initialize`?

Reply:
hmm, ok, makes sense and fine by me, but i'd maybe then rename this from `_initialize` to `_intialize_builtins` or something else a little more descriptive